packages/contracts-bedrock/src/governance/GovernanceToken.sol

```diff
@@ -5,43 +5,165 @@ import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
 import "@openzeppelin/contracts/token/ERC20/extensions/ERC20Burnable.sol";
 import "@openzeppelin/contracts/token/ERC20/extensions/ERC20Votes.sol";
 import "@openzeppelin/contracts/access/Ownable.sol";
+import "src/libraries/Predeploys.sol";
+import "src/governance/GovernanceDelegation.sol";
 
 /// @custom:predeploy 0x4200000000000000000000000000000000000042
 /// @title GovernanceToken
 /// @notice The Optimism token used in governance and supporting voting and delegation. Implements
 ///         EIP 2612 allowing signed approvals. Contract is "owned" by a `MintManager` instance with
 ///         permission to the `mint` function only, for the purposes of enforcing the token
-///         inflation schedule.
+///         inflation schedule. If an account has already been migrated to the GovernanceDelegation
+///         contract, the GovernanceToken contract uses the state from the GovernanceDelegation by
+///         calling the corresponding functions in the GovernanceDelegation contract. If the account
+///         has not been migrated, the GovernanceToken contract uses its own state.
```

Comment:
We will want to make a new contract here that is `GovernanceTokenInterop is GovernanceToken` to not make any breaking changes to the existing codebase. In the L2 genesis script, we will want to use `GovernanceTokenInterop` if `useInterop() == true`. Once we finalize the features and are ready for a release, then we can combine the 2 contracts into 1 and delete `GovernanceTokenInterop`

Reply:
This development pattern is noted in https://github.com/ethereum-optimism/design-docs/blob/2114b1af22a66be19810c2356a48b656ca36308f/smart-contract-feature-development.md